packages/core/src/shared/sam/utils.ts

```diff
@@ -108,6 +108,35 @@ export async function updateRecentResponse(
         getLogger().warn(`sam: unable to save response at key "${key}": %s`, err)
     }
 }
+
+const buildProcessMementoRootKey = 'samcli.build.processes'
+
+/**
+ * Returns true if there's an ongoing build process for the provided template, false otherwise
+ * @Param templatePath The path to the template.yaml file
+ */
+function isBuildInProgress(templatePath: string): boolean {
+    return getRecentResponse(buildProcessMementoRootKey, 'global', templatePath) !== undefined
+}
+
+/**
+ * Throws an error if there's a build in progress for the provided template
+ * @Param templatePath The path to the template.yaml file
+ */
+export function throwIfTemplateIsBeingBuilt(templatePath: string) {
+    if (isBuildInProgress(templatePath)) {
+        throw new ToolkitError('Template is already being built', { code: 'BuildInProgress' })
```

Comment:
Can we make the error code an enum? I think we had a class with different error codes already right, let's add this one in with them

Reply:
Those are specific SAM CLI error patterns. I don't think it makes sense to add this one to them since this error doesn't come directly from SAM CLI.